app/services/remediations_api.rb

```diff
@@ -9,20 +9,29 @@ def initialize(account)
   end
 
   def import_remediations
-    Rule.find_in_batches(batch_size: 100).each do |rules|
-      response = Platform.connection.post(@url) do |req|
-        req.headers['X-RH-IDENTITY'] = @b64_identity
-        req.headers['Content-Type'] = 'application/json'
-        req.body = { 'issues': build_issues_list(rules) }.to_json
-      end
-      update_rules(remediations_available(response))
+    rules_with_profiles.each do |rules|
+      update_rules(remediations_available(remediations_request(rules)))
     end
   rescue Faraday::ClientError => e
     Rails.logger.info("#{e.message} #{e.response}")
   end
 
   private
 
+  def remediations_request(rules)
+    Platform.connection.post(@url) do |req|
+      req.headers['X-RH-IDENTITY'] = @b64_identity
+      req.headers['Content-Type'] = 'application/json'
+      req.body = { 'issues': build_issues_list(rules) }.to_json
+    end
+  end
+
+  def rules_with_profiles
+    Rule.find_in_batches(batch_size: 100).select do |rules|
+      rules.map { |rule| rule.profiles.empty? }
```

Comment:
Notice in Line 41, we call `"ssg:rhel7|#{remove_ref_id_prefix(profile_rules[rule.id])}"` - this method fails if any of the rules does not have a profile.

Reply:
The profile is currently required by remediations to make the request.